### PR TITLE
Code cleanup: remove unused db attrs, dead MoM code

### PR DESCRIPTION
**Pull request by ashwathraop**
#### Describe Bug or Feature

PBS code has a lot of dead code and unused attributes. Most of it accumulated overtime when we stopped supporting  darwin, blue gene, or CSA. 

#### Describe Your Change

Removed the dead code with some refactoring.

#### Attach Test and Valgrind Logs/Output

Initial TH test results:
Description: Tests from given sources on platforms CENTOS7, UBUNTU1804, SUSE15, UBUNTU1604
Platforms: CENTOS7,UBUNTU1804,SUSE15,UBUNTU1604
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|2779|3510|5|0|0|0|3505|

PS: 5 are known failures.

### CODE REVIEW

File: src/cmds/scripts/pbs_init.d.in [hunk 1]
```diff
@@ -113,22 +113,21 @@ lc_host_name()
 #
 : check_started
 check_started() {
+	ps_out=`ps -p $1 -o args 2> /dev/null | tail -1`
```
Comment (hirenvadalia) [inline src/cmds/scripts/pbs_init.d.in:116]: Thank you for formatting this file. One optional comment, can we also change `` to `$()` syntax?

File: src/cmds/scripts/pbs_init.d.in [hunk 2]
```diff
@@ -113,22 +113,21 @@ lc_host_name()
...
-  fi
-
-# strip out everything except executable name
-  prog_name=`echo ${ps_out} | grep -how "$2"`
-  if [ "x${prog_name}" = "x$2" ] ; then
-    return 0;
-  fi
+	# strip out everything except executable name
+	prog_name=`echo ${ps_out} | grep -how "$2"`
+	if [ "x${prog_name}" = "x$2" ]; then
```
Comment (hirenvadalia) [inline src/cmds/scripts/pbs_init.d.in:126]: not sure, but shouldn't this be `==`?

File: src/cmds/scripts/pbs_init.d.in [hunk 3]
```diff
@@ -145,53 +144,53 @@ check_started() {
...
+				pid=${pbs_secondary_comm_pid} ;
+			fi ;;
+
+		*)	echo Invalid PBS daemon name: $1 >&2;
+			return 1;;
+	esac
+
+	if [ -n "${pid}" ]; then
+		if [ "${pid}" -ne -1 ] ; then
+			if check_started "${pid}" "$daemon_name" ; then
```
Comment (hirenvadalia) [inline src/cmds/scripts/pbs_init.d.in:186]: Also can you please change this kinda syntax to `if []; then`? or others to this one. I mean we should use one syntax.

File: src/cmds/scripts/pbs_init.d.in [hunk 4]
```diff
@@ -269,188 +268,183 @@ check_hostname() {
...
+	fi
 
-         if [ -r "${PBS_HOME}/server_priv/qmgr_startup" ]; then
-           echo "pbs_server evaluating ${PBS_HOME}/server_priv/qmgr_startup"
-           ${PBS_EXEC}/bin/qmgr <"${PBS_HOME}/server_priv/qmgr_startup"
-         fi
-      fi
-    fi
+	if [ "${PBS_START_SERVER}" -gt 0 ] ; then
+		if check_prog "server" || check_prog "oldserver" ; then
```
Comment (hirenvadalia) [inline src/cmds/scripts/pbs_init.d.in:407]: I don't think we need `"oldserver"` one now? IIRC, then it was was for our file based server.

File: src/cmds/scripts/pbs_init.d.in [hunk 5]
```diff
@@ -269,188 +268,183 @@ check_hostname() {
...
+						echo "pbs_server startup failed, exit ${ret_val} aborting." >&2
+						return 1
+					fi
+				else
+					echo "pbs_server startup failed, exit ${ret_val} aborting." >&2
+					return 1
+				fi
+			fi
+
+			if [ -r "${PBS_HOME}/server_priv/qmgr_startup" ]; then
```
Comment (hirenvadalia) [inline src/cmds/scripts/pbs_init.d.in:433]: I think we can get rid of this, but can you please check once? last time I looked for this I couldn't find anything in guides.

File: src/include/job.h [hunk 6]
```diff
@@ -552,43 +552,39 @@ struct job {
...
-		char    ji_queue[PBS_MAXQUEUENAME+1];  /* name of current queue */
-		char    ji_destin[PBS_MAXROUTEDEST+1]; /* dest from qmove/route */
+		int	ji_jsversion;				/* job structure version - JSVERSION */
+		int	ji_state;				/* internal copy of state */
+		int	ji_substate;				/* job sub-state */
+		int	ji_svrflags;				/* server flags */
+		time_t  ji_stime;				/* time job started execution */
+
+		char	ji_jobid[PBS_MAXSVRJOBID+1];		/* job identifier */
+		char	ji_fileprefix[PBS_JOBBASE+1];		/* no longer used */
```
Comment (hirenvadalia) [inline src/include/job.h:562]: comment says no longer used? so can we get rid of it?

File: src/include/mom_func.h [hunk 7]
```diff
@@ -266,11 +264,12 @@ extern pre_finish_results_t pre_finish_exec(job *pjob, int do_job_setup_send);
 extern void	finish_exec(job *pjob);
 extern void	exec_bail(job *pjob, int code, char *txt);
 extern int	generate_pbs_nodefile(job *pjob, char *nodefile,
-			int nodefile_sz, char *err_msg, int err_msg_sz);
+	int nodefile_sz, char *err_msg, int err_msg_sz);
```
Comment (hirenvadalia) [inline src/include/mom_func.h:267]: merge this with above line

File: src/include/job.h [hunk 8]
```diff
@@ -600,14 +596,8 @@ struct job {
 	union jobextend {
 		char fill[256];	/* fill to keep same size */
 		struct {
-#if defined(__sgi)
-			jid_t	ji_jid;
-			ash_t	ji_ash;
-#else
 			char	ji_4jid[8];
```
Comment (hirenvadalia) [inline src/include/job.h:599]: can we rename this to more meaning full name? Also please add comments for members which doesn't have

File: src/lib/Libdb/pgsql/db_job.c [hunk 9]
```diff
@@ -315,10 +271,10 @@ static int
 load_job(const  PGresult *res, pbs_db_job_info_t *pj, int row)
 {
 	char *raw_array;
-	static int ji_jobid_fnum, ji_state_fnum, ji_substate_fnum, ji_svrflags_fnum, ji_numattr_fnum,
-	ji_ordering_fnum, ji_priority_fnum, ji_stime_fnum, ji_endtBdry_fnum, ji_queue_fnum, ji_destin_fnum,
-	ji_un_type_fnum, ji_momaddr_fnum, ji_momport_fnum, ji_exitstat_fnum, ji_quetime_fnum, ji_rteretry_fnum,
-	ji_fromsock_fnum, ji_fromaddr_fnum, ji_4jid_fnum, ji_4ash_fnum, ji_credtype_fnum, ji_qrank_fnum,
+	static int ji_jobid_fnum, ji_state_fnum, ji_substate_fnum, ji_svrflags_fnum,
```
Comment (hirenvadalia) [inline src/lib/Libdb/pgsql/db_job.c:274]: make variable declaration one per line please

File: src/lib/Libdb/pgsql/db_resv.c [hunk 10]
```diff
@@ -209,8 +184,7 @@ load_resv(PGresult *res, pbs_db_resv_info_t *presv, int row)
 {
 	char *raw_array;
 	static int ri_resvid_fnum, ri_queue_fnum, ri_state_fnum, ri_substate_fnum, ri_stime_fnum,
-	ri_etime_fnum, ri_duration_fnum, ri_tactive_fnum, ri_svrflags_fnum, ri_numattr_fnum, ri_resvTag_fnum,
-	ri_un_type_fnum, ri_fromsock_fnum, ri_fromaddr_fnum, attributes_fnum;
+	ri_etime_fnum, ri_duration_fnum, ri_tactive_fnum, ri_svrflags_fnum, attributes_fnum;
```
Comment (hirenvadalia) [inline src/lib/Libdb/pgsql/db_resv.c:187]: same as above, and other funcs in this file too.

File: test/fw/ptl/lib/pbs_testlib.py [hunk 11]
```diff
@@ -10422,12 +10422,11 @@ def __insert_jobs_in_db(self, jobs, hostname=None):
                          "attr_resource, attr_value, attr_flags)")
 
         job_stmt = ("INSERT INTO pbs.job (ji_jobid, ji_sv_name, ji_state, "
-                    "ji_substate,ji_svrflags, ji_numattr,"
-                    " ji_ordering, ji_priority, ji_stime, ji_endtbdry, "
-                    "ji_queue, ji_destin, ji_un_type, ji_momaddr, "
-                    "ji_momport, ji_exitstat, ji_quetime, ji_rteretry, "
-                    "ji_fromsock, ji_fromaddr, ji_4jid, ji_4ash, "
-                    "ji_credtype, ji_qrank, ji_savetm, ji_creattm)")
+                    "ji_substate, ji_svrflags, ji_stime, "
```
Comment (hirenvadalia) [inline test/fw/ptl/lib/pbs_testlib.py:10425]: This is not working experimental code, see if you can get rid of this also

File: src/cmds/scripts/pbs_init.d.in (hunk 2 above)
Comment (agrawalravi90) [reply src/cmds/scripts/pbs_init.d.in:126]: it seems like == is a bash specific operator which is an alias of =, = is more portable across different shells, so we should use =:
https://stackoverflow.com/questions/20449543/bash-equality-operators-eq

File: test/fw/ptl/lib/pbs_testlib.py (hunk 11 above)
Comment (ashwathraop) [reply test/fw/ptl/lib/pbs_testlib.py:10425]: This looks like ptl framework code and I am not well versed with this part of the code.

File: src/cmds/scripts/pbs_init.d.in (hunk 2 above)
Comment (hirenvadalia) [reply src/cmds/scripts/pbs_init.d.in:126]: I don't think so @agrawalravi90, we use /bin/bash as shebang line in this file.

Comment (ashwathraop) [reply src/cmds/scripts/pbs_init.d.in:126]: = and == both work for string matching.

File: src/include/job.h (hunk 6 above)
Comment (Bhagat-Rajput) [reply src/include/job.h:562]: If we are going to change/modify job_fix or task_fix structure of job then i guess need to add corresponding changes in pbs_upgrade_job tool to support overlay upgrade feature

Comment (ashwathraop) [reply src/include/job.h:562]: I have another PR in works to refactor job's quick save. I can consider this one in that PR. There is still code using this variable. May be we dont get to that code path anymore.

**Comment by hirenvadalia** [comment]
@ashwathraop changes looks good. Can you please re-run regression?

**Comment by ashwathraop** [comment]
Here are the latest results. They seem to match with failures on master:

Description: Tests from given sources on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|3423|3439|3|3|0|2|3431|

**Review by hirenvadalia — APPROVED** [review]

File: src/server/process_request.c [hunk 12]
```diff
@@ -155,36 +155,25 @@ static void close_quejob(int sfds);
...
-				jobp->ji_extended.ji_ext.ji_credtype = \
-							PBS_CREDTYPE_AES;
-			} else
-				ret = read_cred(jobp, data, dsize);
+	/*
+	 * ensure job's euser exists as this can be called
+	 * from pbs_send_job who is moving a job from a routing
+	 * queue which doesn't have euser set
+	 */
+	if ((jobp->ji_wattr[JOB_ATR_euser].at_flags & ATR_VFLAG_SET) && jobp->ji_wattr[JOB_ATR_euser].at_val.at_str) {
```
Comment (agrawalravi90) [inline src/server/process_request.c:166]: thanks for fixing the formatting!

File: src/resmom/Makefile.am [hunk 13]
```diff
@@ -44,7 +44,7 @@ sbin_PROGRAMS = pbs_mom
 pbs_mom_CPPFLAGS = \
 	-DPBS_MOM \
 	-I$(top_srcdir)/src/include \
-	-I$(top_srcdir)/src/resmom/${PBS_MACH} \
+	-I$(top_srcdir)/src/resmom/linux \
```
[the remark this answers is not in the document]
Comment (ashwathraop) [reply src/resmom/Makefile.am:47]: I guess it's better to keep this way if we decide to add a new platform

File: src/include/pbs_db.h [hunk 14]
```diff
@@ -166,28 +166,21 @@ typedef struct pbs_db_mominfo_time pbs_db_mominfo_time_t;
...
-	char     ji_4ash[8];  	/* extended job save data */
-	INTEGER  ji_credtype; 	/* credential type */
-	INTEGER  ji_qrank;    	/* sort key for db query */
+	INTEGER  ji_un_type;	/* job's queue type */
+	INTEGER  ji_exitstat;	/* job exit status from MOM */
+	BIGINT   ji_quetime;	/* time entered queue */
+	BIGINT   ji_rteretry;	/* route retry time */
+	INTEGER  ji_fromsock;	/* socket job coming over */
+	BIGINT   ji_fromaddr;	/* host job coming from   */
+	char     ji_jid[8];	/* extended job save data */
```
[the remark this answers is not in the document]
Comment (ashwathraop) [reply src/include/pbs_db.h:181]: We use it to capture jobid from cray systems. And it is not used except a few places where we work with cray. So I feel it is okay to keep it the same.

**Comment by ashwathraop** [comment]
> Minor comments. Also, since you are changing the schema, can you please do upgrade testing?

I tried to test and upgrade is already broken and there are plans to change how we upgrade.

**Pull request merged**